File: src/main/java/vorquel/mod/vderps/block/BlockVD.java
package vorquel.mod.vderps.block;

import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;
import net.minecraft.block.Block;
import net.minecraft.block.material.Material;
import net.minecraft.client.renderer.texture.IIconRegister;
import vorquel.mod.vderps.helper.Ref;

public class BlockVD extends Block {
	
	public final String blockName;
	
	public BlockVD(String name, Material material) {
		super(material);
		blockName = name;
		setCreativeTab(Ref.VD_TAB());
	}
	
	public BlockVD(String name) {
		this(name, Material.rock);
	}
	
	@Override
	public String getUnlocalizedName() {
		return String.format("tile.%s", getShortName());
	}
	
	@Override
	@SideOnly(Side.CLIENT)
	public void registerBlockIcons(IIconRegister iconRegister) {
		blockIcon = iconRegister.registerIcon(getShortName());
	}
	
	public String getShortName() {
		return String.format("%s:%s", Ref.MOD_ID().toLowerCase(), blockName);
	}
}

File: src/main/scala/vorquel/mod/vderps/block/BlockVD.scala
package vorquel.mod.vderps.block

import cpw.mods.fml.relauncher.{SideOnly, Side}
import net.minecraft.block.Block
import net.minecraft.block.material.Material
import net.minecraft.client.renderer.texture.IIconRegister
import vorquel.mod.vderps.helper.Ref

class BlockVD(name: String, material: Material) extends Block(material) {
  val blockName: String = name
  setCreativeTab(Ref.VD_TAB)

  def this(name: String) {
    this(name, Material.rock)
  }

  override def getUnlocalizedName: String = String.format("tile.%s", getShortName)

  @SideOnly(Side.CLIENT) override def registerBlockIcons(iconRegister: IIconRegister) {
    blockIcon = iconRegister.registerIcon(getShortName)
  }

  def getShortName: String = String.format("%s:%s", Ref.MOD_ID.toLowerCase, blockName)
}
